src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php

```diff
@@ -0,0 +1,739 @@
+    {
+        $tableId = $this->inputFilterUtil->getFilteredGetInput('tableid', '');
+        $nodeId = $this->inputFilterUtil->getFilteredGetInput('nodeId', '');
+
+        if ('' === $tableId || '' === $nodeId) {
+            return null;
+        }
+
+        $tableEditor = $this->tools->GetTableEditorManager($this->treeTable, $nodeId);
+        $tableEditor->Delete($nodeId);
+
+        return $nodeId;
+    }
+
+    public function connectPageToNode(): ?string
+    {
+        $treeNodeTableName = $this->inputFilterUtil->getFilteredGetInput('table', '');
+        $tableId = $this->tools->GetCMSTableId($treeNodeTableName);
+        $nodeId = $this->inputFilterUtil->getFilteredGetInput('nodeId', '');
+        $currentPageId = $this->inputFilterUtil->getFilteredGetInput('currentPageId', '');
+
+        if ('' === $treeNodeTableName || '' === $nodeId || '' === $currentPageId) {
+            return null;
+        }
+
+        $tableEditor = $this->tools->GetTableEditorManager($treeNodeTableName);
+        $insertSuccessData = $tableEditor->Insert();
+        $recordId = $insertSuccessData->id;
+
+        $recordData = [];
+        $recordData['tableid'] = $tableId;
+        $recordData['id'] = $recordId;
+        $recordData['active'] = '1';
+        $recordData['start_date'] = '0000-00-00 00:00:00';
+        $recordData['end_date'] = '0000-00-00 00:00:00';
+        $recordData['cms_tree_id'] = $nodeId;
+        $recordData['contid'] = $currentPageId;
+        $recordData['tbl'] = 'cms_tpl_page';
+
+        $tableEditor->AllowEditByAll(true);
+        $tableEditor->Save($recordData);
+        $tableEditor->AllowEditByAll(false);
+
+        return $nodeId;
+    }
+
+    public function disconnectPageFromNode(): ?string
+    {
+        $treeNodeTableName = $this->inputFilterUtil->getFilteredGetInput('table', '');
+        $nodeId = $this->inputFilterUtil->getFilteredGetInput('nodeId', '');
+        $currentPageId = $this->inputFilterUtil->getFilteredGetInput('currentPageId', '');
+
+        if ('' === $treeNodeTableName || '' === $nodeId || '' === $currentPageId) {
+            return null;
+        }
+
+        $query = 'SELECT *
+                FROM `cms_tree_node`
+               WHERE `cms_tree_node`.`contid` = '.$this->dbConnection->quote($currentPageId).'
+                 AND `cms_tree_node`.`cms_tree_id` = '.$this->dbConnection->quote($nodeId)."
+                 AND `cms_tree_node`.`tbl` = 'cms_tpl_page'
+               ";
+        $nodePageConnectionList = \TdbCmsTreeNodeList::GetList($query);
+        while ($nodePageConnection = $nodePageConnectionList->Next()) {
+            $tableEditor = $this->tools->GetTableEditorManager($treeNodeTableName, $nodePageConnection->id);
+            $tableEditor->Delete($nodePageConnection->id);
+        }
+
+        return $nodeId;
+    }
+
+    /**
+     * Renders all children of a node.
+     * Is called via ajax and converted to JSON.
+     */
+    public function getTreeNodesJson(): array
```

Comment:
Plural Nodes: Yes, it loads the requested node and its children.
At first the return value was just a json, but now the json has array brackets [] around it because jstree wants it that way. So I changed the method name form getTreeNodesJson() to getTreeNodes().